src/artemis/tests/test_fast_grid_scan_plan.py

```diff
@@ -60,136 +64,126 @@ def test_ispyb_params_update_from_ophyd_devices_correctly():
-    mock_ispyb_get_time.return_value = DUMMY_TIME_STRING
-    mock_ispyb_update_time_and_status.return_value = None
-
-    with pytest.raises(Exception) as excinfo:
-        with patch(
-            "artemis.fast_grid_scan_plan.NexusWriter",
-            side_effect=Exception("mocked error"),
-        ):
-            list(run_gridscan(*dummy_3d_gridscan_args))
-
-    expected_error_message = "mocked error"
-    assert str(excinfo.value) == expected_error_message
-
-    mock_ispyb_update_time_and_status.assert_has_calls(
-        [call(DUMMY_TIME_STRING, BAD_ISPYB_RUN_STATUS, id, dcg_id) for id in dc_ids]
+    RE = RunEngine({})
+    params = FullParameters()
+    communicator = FGSCommunicator()
+    communicator.xray_centre_motor_position = Point3D(1, 2, 3)
+    FakeComposite = make_fake_device(FGSComposite)
+    FakeEiger = make_fake_device(EigerDetector)
+    RE(
+        run_gridscan_and_move(
+            FakeComposite("test", name="fakecomposite"),
+            FakeEiger(params.detector_params),
+            params,
+            communicator,
+        )
     )
-    assert mock_ispyb_update_time_and_status.call_count == len(dc_ids)
-
+    run_gridscan.assert_called_once()
+    move_xyz.assert_called_once()
 
-@patch("artemis.fast_grid_scan_plan.run_start")
-@patch("artemis.fast_grid_scan_plan.run_end")
-@patch("artemis.fast_grid_scan_plan.wait_for_result")
-@patch("artemis.fast_grid_scan_plan.StoreInIspyb3D.store_grid_scan")
-@patch("artemis.fast_grid_scan_plan.StoreInIspyb3D.get_current_time_string")
-@patch(
-    "artemis.fast_grid_scan_plan.StoreInIspyb3D.update_grid_scan_with_end_time_and_status"
-)
-def test_fgs_raising_no_exception_results_in_good_run_status_in_ispyb(
-    mock_ispyb_update_time_and_status: MagicMock,
-    mock_ispyb_get_time: MagicMock,
-    mock_ispyb_store_grid_scan: MagicMock,
-    wait_for_result: MagicMock,
-    run_end: MagicMock,
-    run_start: MagicMock,
-    dummy_3d_gridscan_args,
-):
-    dc_ids = [1, 2]
-    dcg_id = 4
 
-    mock_ispyb_store_grid_scan.return_value = [dc_ids, None, dcg_id]
-    mock_ispyb_get_time.return_value = DUMMY_TIME_STRING
-    mock_ispyb_update_time_and_status.return_value = None
+@pytest.fixture
+def dummy_3d_gridscan_args():
+    params = FullParameters()
+    params.grid_scan_params.z_steps = 2
 
-    with patch("artemis.fast_grid_scan_plan.NexusWriter"):
-        list(run_gridscan(*dummy_3d_gridscan_args))
+    FakeFGSComposite = make_fake_device(FGSComposite)
+    fgs_composite: FGSComposite = FakeFGSComposite(name="fgs", insertion_prefix="")
 
-    mock_ispyb_update_time_and_status.assert_has_calls(
-        [call(DUMMY_TIME_STRING, GOOD_ISPYB_RUN_STATUS, id, dcg_id) for id in dc_ids]
+    FakeEiger = make_fake_device(EigerDetector)
+    eiger: EigerDetector = FakeEiger(
+        detector_params=params.detector_params, name="eiger"
     )
-    assert mock_ispyb_update_time_and_status.call_count == len(dc_ids)
+
+    return fgs_composite, eiger, params
```

Comment:
Should: You can remove this fixture as it is no longer used. However, you could also probably use it in some of your tests to remove boilerplate such as making the fake devices. Up to you.

Reply:
ok, removed